File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoPulseFragmentModule.java
package mb00.android.codehub.ui.repo.inject;

import dagger.Module;
import dagger.Provides;
import mb00.android.codehub.ui.repo.interactor.RepoInteractor;
import mb00.android.codehub.ui.repo.router.RepoRouter;
import mb00.android.codehub.ui.repo.viewmodel.RepoPulseViewModel;


@Module
public class RepoPulseFragmentModule {

    @Provides
    RepoPulseViewModel provideRepoPulseViewModel(RepoInteractor interactor, RepoRouter router) {
        return new RepoPulseViewModel(interactor, router);
    }

}

File: app/src/main/java/mb00/android/codehub/ui/repo/inject/RepoPulseFragmentModule.kt
package mb00.android.codehub.ui.repo.inject

import dagger.Module
import dagger.Provides
import mb00.android.codehub.ui.repo.interactor.RepoInteractor
import mb00.android.codehub.ui.repo.router.RepoRouter
import mb00.android.codehub.ui.repo.viewmodel.RepoPulseViewModel


@Module
class RepoPulseFragmentModule {

    @Provides
    internal fun provideRepoPulseViewModel(interactor: RepoInteractor, router: RepoRouter): RepoPulseViewModel {
        return RepoPulseViewModel(interactor, router)
    }

}
